docs/DeveloperGuide.md

```diff
@@ -36,3 +44,64 @@
 ## Instructions for manual testing
 
 {Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+
+## Parser Component
+
+Here is a general guide of how Parse Class looks like in sequence diagram.
+
+![Parse.png](diagram%2FParse.png)
+
+Here is a class diagram for the method changeTaskTiming method in Parser Class.
+
+![changeTaskTimingClassDiagram.png](diagram%2FchangeTaskTimingClassDiagram.png)
+
+The following sequence diagram of changeTaskTiming shows the interactions with other objects.
+
+![changeTaskTiming_Sequence_Diagram.png](diagram%2FchangeTaskTiming_Sequence_Diagram.png)
+
+Here is how this method works:
+
+1. The Parser calls the changeTaskTiming method with a command and the UserList. It calls InputValidator.
+
+2. The InputValidator validates the command, ensuring that it meets the required format in days and index.
+
+3. If validation is successful, the User calls the timetable and calls its changeFlexibleTaskTiming method with the specified parameters.
+
+4. The Timetable updates the timing of the flexible task.
+
+5. If successful, the InputValidator sends a success message back to the Parser. Otherwise, it throws a RuntimeException.
+
+## Timetable Component
+Here is a sequence diagram of changeTaskType method in Timetable class. It shows the interactions with other objects.
+![changeTaskType.png](diagram%2FchangeTaskType.png)
+
+Here is how this method works:
```

Comment:
Good breakdown of the changeTaskType method. However, should there be the other methods of the Timetable class?